crest/Assets/Crest/Crest/Scripts/Spline/SplinePoint.cs

```diff
@@ -0,0 +1,178 @@
+            }
+            Gizmos.color = Color.white;
+        }
+#endif
+    }
+
+#if UNITY_EDITOR
+    public interface IReceiveSplinePointOnDrawGizmosSelectedMessages
+    {
+        void OnSplinePointDrawGizmosSelected(SplinePoint point);
+    }
+
+    [CustomEditor(typeof(SplinePoint))]
+    public class SplinePointEditor : Editor
+    {
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            var thisSP = target as SplinePoint;
+            var thisIdx = thisSP.transform.GetSiblingIndex();
+
+            var parent = thisSP.transform.parent;
+            if (parent == null || parent.GetComponent<Spline>() == null)
+            {
+                EditorGUILayout.HelpBox("Spline component must be present on parent of this GameObject.", MessageType.Error);
+                return;
+            }
+
+            GUILayout.Label("Selection", EditorStyles.boldLabel);
+
+            GUILayout.BeginHorizontal();
+            if (thisIdx == 0) GUI.enabled = false;
+            if (GUILayout.Button("Select previous"))
+            {
+                Selection.activeObject = parent.GetChild(thisIdx - 1).gameObject;
+            }
+            GUI.enabled = true;
+            if (thisIdx == parent.childCount - 1) GUI.enabled = false;
+            if (GUILayout.Button("Select next"))
+            {
+                Selection.activeObject = parent.GetChild(thisIdx + 1).gameObject;
+            }
+            GUI.enabled = true;
+            GUILayout.EndHorizontal();
+
+            if (GUILayout.Button("Select spline"))
+            {
+                Selection.activeObject = parent.gameObject;
+            }
+
+            GUILayout.Label("Spline actions", EditorStyles.boldLabel);
+
+            GUILayout.BeginHorizontal();
+            string label;
+
+            label = thisIdx == 0 ? "Add before (extend)" : "Add before";
+            if (GUILayout.Button(label))
+            {
+                var newPoint = CreateNewSP();
+                newPoint.transform.parent = parent;
+                var newIdx = thisIdx;
+                newPoint.transform.SetSiblingIndex(newIdx);
+
+                if (thisIdx > 0)
+                {
+                    var beforeNewPoint = parent.GetChild(thisIdx - 1);
+                    newPoint.transform.position = Vector3.Lerp(beforeNewPoint.position, thisSP.transform.position, 0.5f);
+                }
+                else if (parent.childCount > 2)
+                {
+                    newPoint.transform.position = 2f * parent.GetChild(newIdx + 1).position - parent.GetChild(newIdx + 2).position;
+                }
+
+                Selection.activeObject = newPoint;
+            }
+
+            label = (thisIdx == parent.childCount - 1 || parent.childCount == 0) ? "Add after (extend)" : "Add after";
+            if (GUILayout.Button(label))
+            {
+                Selection.activeObject = AddSplinePointAfter(parent, thisIdx);
```

Comment:
Needs to register the created object with edit history so undo removes the new spline point instead of just changing the selection back to the previous point:
```csharp
Undo.RegisterCreatedObjectUndo(Selection.activeObject, "Add Crest Spline Point");
```

This could be done in the function that creates it. But works here too.

Reply:
done